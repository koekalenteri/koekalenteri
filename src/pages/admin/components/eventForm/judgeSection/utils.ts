import type { DeepPartial, EventClass, EventType, Judge, PublicJudge } from '../../../../../types'
import type { PartialEvent } from '../../EventForm'

import { isSameDay } from 'date-fns'

export type PartialPublicJudge = Partial<PublicJudge>
export type PartialPublicJudgeValue = PartialPublicJudge | PartialPublicJudge[]

export const filterJudges = (
  judges: Judge[],
  eventJudges: PublicJudge[],
  id: number | undefined,
  eventType?: EventType
) =>
  judges
    .filter((j) => !eventType?.official || j.eventTypes.includes(eventType.eventType))
    .filter((j) => j.id === id || !eventJudges.find((ej) => ej.id === j.id))

export const hasJudge = (c: DeepPartial<EventClass>, id?: number) =>
  Array.isArray(c.judge) ? c.judge.find((j) => j.id === id) : c.judge?.id === id

export const filterClassesByJudgeId = (classes?: EventClass[], id?: number) => classes?.filter((c) => hasJudge(c, id))

export const toArray = (j?: PartialPublicJudge): PartialPublicJudge[] => (j ? [j] : [])
export const makeArray = (j?: PartialPublicJudgeValue) => (Array.isArray(j) ? [...j] : toArray(j))
export const selectJudge = (j?: PartialPublicJudgeValue, judge?: PublicJudge): PartialPublicJudge[] => {
  const a = makeArray(j)
  if (judge) {
    const index = a.findIndex((cj) => cj.id === judge.id)
    if (index === -1) {
      a.push(judge)
    } else {
      a[index] = judge
    }
  }
  return a
}

export const removeJudge = (j?: PartialPublicJudgeValue, id?: number): PartialPublicJudge[] => {
  const a = makeArray(j)
  return a.filter((cj) => cj.id !== id)
}

export const updateJudge = (
  event: PartialEvent,
  id: number | undefined,
  judge: PublicJudge | undefined,
  values?: DeepPartial<EventClass>[]
) => {
  const isSelected = (c: DeepPartial<EventClass>) =>
    values?.find((v) => event && isSameDay(v.date ?? event.startDate, c.date ?? event.startDate) && v.class === c.class)
  return event.classes?.map((c) => ({
    ...c,
    judge: isSelected(c) ? selectJudge(c.judge, judge) : removeJudge(c.judge, id),
  }))
}
