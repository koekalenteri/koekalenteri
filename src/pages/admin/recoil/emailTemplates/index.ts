export * from './actions'
export * from './atoms'
export * from './atoms2'
export * from './effects'
export * from './selectors'
