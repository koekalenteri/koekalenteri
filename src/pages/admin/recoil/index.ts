export * from './emailTemplates'
export * from './events'
export * from './eventTypes'
export * from './officials'
export * from './organizers'
export * from './registrations'
export * from './user'
