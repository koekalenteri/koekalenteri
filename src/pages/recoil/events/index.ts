export * from './atoms'
export * from './effects'
export * from './filters'
export * from './selectors'
export * from './types'
