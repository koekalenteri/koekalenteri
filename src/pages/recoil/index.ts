export * from './effects'
export * from './events'
export * from './judges'
export * from './registration'
export * from './user'
