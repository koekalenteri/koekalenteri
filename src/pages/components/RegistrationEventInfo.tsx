import type { PublicDogEvent } from '../../types'

import { useTranslation } from 'react-i18next'
import PictureAsPdfOutlined from '@mui/icons-material/PictureAsPdfOutlined'
import Grid from '@mui/material/Grid'
import Link from '@mui/material/Link'
import Paper from '@mui/material/Paper'

import useEventStatus from '../../hooks/useEventStatus'
import { judgeName } from '../../lib/judge'
import { printContactInfo } from '../../lib/utils'
import { API_BASE_URL } from '../../routeConfig'

import CollapsibleSection from './CollapsibleSection'
import CostInfo from './CostInfo'
import { PriorityChips } from './PriorityChips'

interface Props {
  readonly event: PublicDogEvent
  readonly invitationAttachment?: string | undefined
}

export default function RegistrationEventInfo({ event, invitationAttachment }: Props) {
  const { t } = useTranslation()
  const status = useEventStatus(event)
  const title = `${event.eventType} ${t('dateFormat.datespan', { start: event.startDate, end: event.endDate })} ${
    event.location + (event.name ? ` (${event.name})` : '')
  }`

  return (
    <Paper sx={{ p: 1, mb: 1, width: '100%' }} elevation={2}>
      <CollapsibleSection title={title} border={false}>
        <Grid
          container
          justifyContent="space-between"
          alignItems="flex-start"
          sx={{ '& .MuiGrid-item': { overflow: 'hidden', textOverflow: 'ellipsis' } }}
        >
          <Grid item container sm={12} md columnSpacing={1}>
            <Grid item xs={4}>
              {t('entryTime')}:
            </Grid>
            <Grid item xs={8} sx={{ '& .info': { px: 1 } }}>
              <b>{t('dateFormat.datespan', { start: event.entryStartDate, end: event.entryEndDate })}</b>&nbsp;
              <span className="info">{status}</span>
            </Grid>
            <Grid item xs={4}>
              {t('event.organizer')}:
            </Grid>
            <Grid item xs={8}>
              {event.organizer?.name}
            </Grid>
            <Grid item xs={4}>
              {t('event.judges')}:
            </Grid>
            <Grid item xs={8}>
              {event.judges.map((j) => judgeName(j, t)).join(', ')}
            </Grid>
            {printContactInfo(event.contactInfo?.official) ? (
              <>
                <Grid item xs={4}>
                  {t('event.official')}:
                </Grid>
                <Grid item xs={8}>
                  {printContactInfo(event.contactInfo?.official)}
                </Grid>
              </>
            ) : null}
            {printContactInfo(event.contactInfo?.secretary) ? (
              <>
                <Grid item xs={4}>
                  {t('event.secretary')}:
                </Grid>
                <Grid item xs={8}>
                  {printContactInfo(event.contactInfo?.secretary)}
                </Grid>
              </>
            ) : null}
          </Grid>
          <Grid item container sm={12} md columnSpacing={1}>
            <Grid item xs={4}>
              {t('event.cost')}:
            </Grid>
            <Grid item xs={8}>
              <CostInfo event={event} />
            </Grid>
            {event.priority ? (
              <>
                <Grid item xs={4}>
                  {t('event.priority')}:
                </Grid>
                <Grid item xs={8}>
                  <PriorityChips priority={event.priority} />
                </Grid>
              </>
            ) : null}
            {event.description ? (
              <>
                <Grid item xs={4}>
                  {t('event.description')}:
                </Grid>
                <Grid item xs={8}>
                  {event.description}
                </Grid>
              </>
            ) : null}
            {invitationAttachment && event.state === 'invited' ? (
              <>
                <Grid item xs={4}>
                  {t('event.attachments')}:
                </Grid>
                <Grid item xs={8}>
                  <PictureAsPdfOutlined fontSize="small" sx={{ verticalAlign: 'middle', pr: 0.5 }} />
                  <Link
                    href={`${API_BASE_URL}/file/${invitationAttachment}/kutsu.pdf`}
                    rel="noopener"
                    target="_blank"
                    type="application/pdf"
                    variant="caption"
                  >
                    Kutsu.pdf
                  </Link>
                </Grid>
              </>
            ) : null}
          </Grid>
        </Grid>
      </CollapsibleSection>
    </Paper>
  )
}
