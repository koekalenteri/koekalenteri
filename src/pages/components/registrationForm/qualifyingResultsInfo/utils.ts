import type { EventResultRequirement, EventResultRequirements, EventResultRequirementsByDate } from '../../../../rules'
import type { ManualTestResult, QualifyingResult } from '../../../../types'

import { nanoid } from 'nanoid'

import { unique } from '../../../../lib/utils'
import { objectContains } from '../validation'

const asArray = (v: EventResultRequirements | EventResultRequirement) => (Array.isArray(v) ? v : [v])

function findFirstMissing(requirements: EventResultRequirementsByDate | undefined, results: QualifyingResult[]) {
  if (!requirements || typeof requirements.rules === 'function') {
    return []
  }
  for (const rule of requirements.rules) {
    for (const opt of asArray(rule)) {
      const { count, ...rest } = opt
      if (results.filter((r) => objectContains(r, rest)).length < count) {
        return rest
      }
    }
  }
}

export function availableTypes(requirements?: EventResultRequirementsByDate, eventType?: string) {
  if (!requirements || typeof requirements.rules === 'function') {
    switch (eventType) {
      case 'NOME-B SM':
        return ['NOME-B']
      case 'NOME-A SM':
        return ['NOME-A', 'NOME-A KV']
      case 'NOWT SM':
        return ['NOWT']
    }
    return []
  }
  return unique(requirements.rules.flatMap((rule) => asArray(rule).map((opt) => opt.type)))
}

const availableSMResults = (type?: string, eventType?: string, hasKVA?: boolean) => {
  switch (eventType) {
    case 'NOME-B SM':
      return [hasKVA ? undefined : 'FI KVA-B', 'VOI1', 'VOI2', 'VOI3'].filter(Boolean)
    case 'NOWT SM':
      return [hasKVA ? undefined : 'FI KVA-WT', 'VOI1', 'VOI2', 'VOI3'].filter(Boolean)
    case 'NOME-A SM':
      if (type === 'NOME-A')
        return [hasKVA ? undefined : 'FI KVA-FT', 'A1 CERT', 'A1 RES-CERT', 'A1', 'A2', 'A3'].filter(Boolean)

      if (type === 'NOME-A KV') return ['EXC CACIT', 'EXC RES-CACIT', 'EXC', 'VG', 'G']
  }
  return []
}

export function availableResults(
  requirements?: EventResultRequirementsByDate,
  type?: string,
  eventType?: string,
  existingResults?: ManualTestResult[]
) {
  if (!requirements || typeof requirements.rules === 'function') {
    if (eventType?.includes(' SM')) {
      const hasKVA = existingResults?.some((r) => r.result.startsWith('FI KVA'))

      return availableSMResults(type, eventType, hasKVA)
    }
    return []
  }
  return unique(
    requirements.rules.flatMap((rule) =>
      asArray(rule)
        .filter((opt) => !type || opt.type === type)
        .map((opt) => (opt.cert ? 'CERT' : opt.result))
    )
  )
}

export function createMissingResult(
  requirements: EventResultRequirementsByDate | undefined,
  results: Array<QualifyingResult | ManualTestResult>,
  regNo: string
): ManualTestResult {
  const rule = findFirstMissing(requirements, results)
  return {
    id: nanoid(10),
    regNo,
    date: new Date(),
    official: false,
    qualifying: true,
    type: '',
    judge: '',
    location: '',
    result: '',
    class: '',
    ...rule,
  }
}

export function resultBorderColor(qualifying: boolean | undefined) {
  if (qualifying === true) {
    return 'success.light'
  }
  if (qualifying === false) {
    return 'error.main'
  }
}

export function getResultId(result: ManualTestResult | QualifyingResult) {
  if ('id' in result) {
    return result.id
  }
  return result.type + result.class + result.date.toISOString()
}
