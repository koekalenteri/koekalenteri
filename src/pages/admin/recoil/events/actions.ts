import type { DogEvent } from '../../../../types'

import { useTranslation } from 'react-i18next'
import { useNavigate } from 'react-router-dom'
import { addDays, differenceInDays } from 'date-fns'
import { diff } from 'deep-object-diff'
import { useSnackbar } from 'notistack'
import { useRecoilState, useRecoilValue, useSetRecoilState } from 'recoil'

import { copyEventWithRegistrations, putEvent } from '../../../../api/event'
import { sanitizeDogEvent } from '../../../../lib/event'
import { Path } from '../../../../routeConfig'
import { eventsAtom, idTokenAtom, userSelector } from '../../../recoil'

import {
  adminEventIdAtom,
  adminNewEventAtom,
  newEventEntryEndDate,
  newEventEntryStartDate,
  newEventStartDate,
} from './atoms'
import { adminCurrentEventSelector } from './selectors'

export const useAdminEventActions = () => {
  const token = useRecoilValue(idTokenAtom)
  const user = useRecoilValue(userSelector)
  const setAdminEventId = useSetRecoilState(adminEventIdAtom)
  const [currentAdminEvent, setCurrentAdminEvent] = useRecoilState(adminCurrentEventSelector)
  const setNewEvent = useSetRecoilState(adminNewEventAtom)
  const [publicEvents, setPublicEvents] = useRecoilState(eventsAtom)
  const { enqueueSnackbar } = useSnackbar()
  const { t } = useTranslation()
  const navigate = useNavigate()

  return {
    copyCurrent,
    copyCurrentTest,
    deleteCurrent,
    save,
  }

  function updatePublicEvents(event: DogEvent, remove?: boolean): void {
    if (event.id) {
      const index = publicEvents.findIndex((e) => e.id === event.id)
      if (index >= 0) {
        const newEvents = [...publicEvents]
        if (remove) {
          newEvents.splice(index, 1)
        } else {
          newEvents.splice(index, 1, sanitizeDogEvent(event))
        }
        setPublicEvents(newEvents)
      } else if (!remove) {
        setPublicEvents([...publicEvents, sanitizeDogEvent(event)])
      }
    }
  }

  function copyCurrent() {
    if (!currentAdminEvent) {
      return
    }
    const copy = structuredClone(currentAdminEvent)
    copy.id = ''
    copy.name = 'Kopio - ' + (copy.name ?? '')
    copy.state = 'draft'
    copy.entries = 0
    copy.classes.forEach((c) => {
      c.entries = c.members = 0
      if (c.date) {
        c.date = addDays(newEventStartDate, differenceInDays(copy.startDate, c.date))
      }
      delete c.state
    })

    const days = differenceInDays(copy.endDate, copy.startDate)
    copy.startDate = newEventStartDate
    copy.endDate = addDays(newEventStartDate, days)
    copy.entryStartDate = newEventEntryStartDate
    copy.entryEndDate = newEventEntryEndDate

    delete copy.kcId
    delete copy.entryOrigEndDate

    setNewEvent(copy)
    navigate(Path.admin.newEvent)
  }

  async function copyCurrentTest() {
    if (!currentAdminEvent) {
      return
    }
    const saved = await copyEventWithRegistrations(currentAdminEvent.id, token)
    setAdminEventId(saved.id)
    setCurrentAdminEvent(saved)
    updatePublicEvents(saved)
    return saved
  }

  async function save(event: Partial<DogEvent>): Promise<DogEvent | undefined> {
    let changes: Partial<DogEvent> = event
    if (event.id && event.id === currentAdminEvent?.id) {
      const changedKeys = Object.keys(diff(currentAdminEvent, event))
      changes = { id: event.id }
      for (const key of changedKeys) {
        //@ts-expect-error typescript is just so stupid sometimes
        changes[key] = event[key]
      }
    }
    console.log(changes)
    const saved = await putEvent(changes, token)
    setAdminEventId(saved.id)
    setCurrentAdminEvent(saved)
    updatePublicEvents(saved)

    return saved
  }

  async function deleteCurrent() {
    if (!currentAdminEvent || currentAdminEvent.deletedAt) {
      return
    }

    await save({
      ...currentAdminEvent,
      deletedAt: new Date(),
      deletedBy: user?.name ?? user?.email,
    })

    updatePublicEvents(currentAdminEvent, true)

    enqueueSnackbar(t('deleteEventComplete'), { variant: 'info' })
  }
}
