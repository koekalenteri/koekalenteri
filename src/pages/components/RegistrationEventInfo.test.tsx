import type { ReactNode } from 'react'
import type { PublicConfirmedEvent } from '../../types'

import { Suspense } from 'react'
import { ThemeProvider } from '@mui/material'
import { LocalizationProvider } from '@mui/x-date-pickers'
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns'
import { render, screen } from '@testing-library/react'
import { RecoilRoot } from 'recoil'

import { eventWithStaticDates } from '../../__mockData__/events'
import theme from '../../assets/Theme'
import { locales } from '../../i18n'
import { flushPromises } from '../../test-utils/utils'

import RegistrationEventInfo from './RegistrationEventInfo'

jest.mock('../../api/event')
jest.mock('../../api/judge')

function Wrapper({ children }: { readonly children: ReactNode }) {
  return (
    <ThemeProvider theme={theme}>
      <LocalizationProvider dateAdapter={AdapterDateFns} adapterLocale={locales.fi}>
        <RecoilRoot>
          <Suspense fallback={<div>loading...</div>}>{children}</Suspense>
        </RecoilRoot>
      </LocalizationProvider>
    </ThemeProvider>
  )
}

describe('RegistrationEventInfo', () => {
  beforeAll(() => jest.useFakeTimers())
  afterEach(() => jest.runOnlyPendingTimers())
  afterAll(() => jest.useRealTimers())

  it('renders', async () => {
    const { container } = render(
      <RegistrationEventInfo event={eventWithStaticDates} invitationAttachment={undefined} />,
      { wrapper: Wrapper }
    )
    await flushPromises()

    expect(screen.queryByText('event.official:')).toBeInTheDocument()
    expect(screen.queryByText('Teemu Toimitsija, 040-official, official@example.com')).toBeInTheDocument()
    expect(screen.queryByText('event.secretary:')).toBeInTheDocument()
    expect(screen.queryByText('Siiri Sihteeri, 040-secretary, secretary@example.com')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('does not render empty contact info', async () => {
    const event: PublicConfirmedEvent = {
      ...eventWithStaticDates,
      contactInfo: { official: {} },
    }
    const { container } = render(<RegistrationEventInfo event={event} invitationAttachment={undefined} />, {
      wrapper: Wrapper,
    })
    await flushPromises()

    expect(screen.queryByText('event.official:')).toBeNull()
    expect(screen.queryByText('event.secretary:')).toBeNull()
    expect(container).toMatchSnapshot()
  })
})
