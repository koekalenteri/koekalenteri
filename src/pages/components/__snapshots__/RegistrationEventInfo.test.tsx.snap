// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationEventInfo does not render empty contact info 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1p4v111-MuiPaper-root"
    style="--Paper-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiBox-root css-i357mj"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-1g8s8mk-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
          data-testid="KeyboardArrowDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
          />
        </svg>
      </button>
      <div
        class="MuiBox-root css-1qbrvqz"
      >
        <div
          class="MuiBox-root css-1jw9kr7"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
          >
            NOU dateFormat.datespan start, end test location (test)
          </p>
          <p
            class="MuiFormHelperText-root css-ml08u8-MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-wz2cs5-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiBox-root css-e1h6et"
              >
                <div
                  class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-15pm0jo-MuiGrid2-root"
                >
                  <div
                    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-sm-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 css-ywssai-MuiGrid2-root"
                  >
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        entryTime
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        <b>
                          dateFormat.datespan start, end
                        </b>
                         
                      </div>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.organizer
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        Suomen Noutajakoirajärjestö ry
                      </div>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.judges
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        Tuomari 1
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-sm-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 css-ywssai-MuiGrid2-root"
                  >
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.cost
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        123 €event.costMember 123 €
                      </div>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.priority
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        <div
                          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-lkqxhd-MuiChip-root"
                        >
                          <span
                            class="MuiChip-label MuiChip-labelSmall css-3fh3fk-MuiChip-label"
                          >
                            priority.members
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.description
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        test
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RegistrationEventInfo renders 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-1p4v111-MuiPaper-root"
    style="--Paper-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiBox-root css-i357mj"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-1g8s8mk-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
          data-testid="KeyboardArrowDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
          />
        </svg>
      </button>
      <div
        class="MuiBox-root css-1qbrvqz"
      >
        <div
          class="MuiBox-root css-1jw9kr7"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1edfpdg-MuiTypography-root"
          >
            NOU dateFormat.datespan start, end test location (test)
          </p>
          <p
            class="MuiFormHelperText-root css-ml08u8-MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-wz2cs5-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiBox-root css-e1h6et"
              >
                <div
                  class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-15pm0jo-MuiGrid2-root"
                >
                  <div
                    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-sm-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 css-ywssai-MuiGrid2-root"
                  >
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        entryTime
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        <b>
                          dateFormat.datespan start, end
                        </b>
                         
                      </div>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.organizer
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        Suomen Noutajakoirajärjestö ry
                      </div>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.judges
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        Tuomari 1
                      </div>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.official
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        Teemu Toimitsija, 040-official, official@example.com
                      </div>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.secretary
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        Siiri Sihteeri, 040-secretary, secretary@example.com
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-sm-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 css-ywssai-MuiGrid2-root"
                  >
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.cost
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        123 €event.costMember 123 €
                      </div>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.priority
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        <div
                          class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-lkqxhd-MuiChip-root"
                        >
                          <span
                            class="MuiChip-label MuiChip-labelSmall css-3fh3fk-MuiChip-label"
                          >
                            priority.members
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-1vng41o-MuiGrid2-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-caption css-1gju9dp-MuiTypography-root"
                      >
                        event.description
                      </span>
                    </div>
                    <div
                      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-7 css-teo3zo-MuiGrid2-root"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-body2 css-1nmqws6-MuiTypography-root"
                      >
                        test
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
